services/merak-topo/handler/handler.go

```diff
@@ -224,31 +200,54 @@ func Create(k8client *kubernetes.Clientset, topo_id string, aca_num uint32, rack
 		}
 	}
 
-	go Topo_deploy(k8client, aca_image, ovs_image, topo, aca_parameters)
+	if namespace != "default" {
+		nsSpec := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+		k8client.CoreV1().Namespaces().Create(context.Background(), nsSpec, metav1.CreateOptions{})
+		utils.Logger.Info("request DEPLOY", "create k8s cluster namespace for new topology deployment", namespace)
+	}
 
-	return nil
+	go Topo_deploy(k8client, aca_image, ovs_image, topo, aca_parameters, topoPrefix, namespace)
+
+	elaps2 := time.Since(start1)
+
+	if err_flag == 1 {
+		err_return = errs
+		utils.Logger.Info("request DEPLOY", "Fails: check merak-topo logs", elaps2)
+	} else {
+		err_return = nil
+		utils.Logger.Info("request DEPLOY", "Complete: Return compute nodes and k8s cluster information to Scenario Manager (in second)", elaps2)
+	}
+
+	return err_return
 
 }
 
-func UpdateComputenodeInfo(k8client *kubernetes.Clientset, topo_id string) error {
+func UpdateComputenodeInfo(k8client *kubernetes.Clientset, topoPrefix string, namespace string) error {
+
+	err_flag := 0
+	errs := errors.New("fails to update compute nodes and k8s node info in DB")
+	var err_return error
+
 	start_time := time.Now()
-	topo, err := database.FindTopoEntity(topo_id, "")
+	topo, err := database.FindTopoEntity(topoPrefix, "")
 
 	if err != nil {
-		return fmt.Errorf("update compute node info: query topology_id error %s", err)
+		utils.Logger.Warn("can't find topology info in DB", topoPrefix, err.Error())
 	}
 
 	k8s_nodes, err1 := k8client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err1 != nil {
-		return fmt.Errorf("UpdateComputenodeInfo: failed to list k8s host nodes info %s", err1)
+		utils.Logger.Error("can't check k8s cluster ", "corev1", err1.Error())
+		err_flag = 1
 	}
 
 	for _, s := range k8s_nodes.Items {
```

Comment:
@zzengcs same here. If `err_flag=`, `k8s_nodes` will be nil or empty and the following `for` loop will complain about it.

Reply:
addressed-- return err when checking k8s cluster worker nodes does not work